types/hapi/definitions/server/server-state-options.d.ts

```diff
@@ -0,0 +1,61 @@
+import {Request} from "hapi";
+
+/**
+ * Optional cookie settings
+ * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#-serverstatename-options)
+ */
+export interface ServerStateCookieOptions {
+    /** time-to-live in milliseconds. Defaults to null (session time-life - cookies are deleted when the browser is closed). */
+    ttl?: number | null;
+    /** sets the 'Secure' flag. Defaults to true. */
+    isSecure?: boolean;
+    /** sets the 'HttpOnly' flag. Defaults to true. */
+    isHttpOnly?: boolean;
+    /**
+     * sets the 'SameSite' flag. The value must be one of:
+     *  * false - no flag.
+     *  * 'Strict' - sets the value to 'Strict' (this is the default value).
+     *  * 'Lax' - sets the value to 'Lax'.
+     */
+    isSameSite?: false | 'Strict' | 'Lax';
+    /** the path scope. Defaults to null (no path). */
+    path?: string | null;
+    /** the domain scope. Defaults to null (no domain). */
+    domain?: string | null;
+    /**
+     * if present and the cookie was not received from the client or explicitly set by the route handler, the
+     * cookie is automatically added to the response with the provided value. The value can be
+     * a function with signature async function(request) where:
+     */
+    autoValue?(request: Request): void;
+    /**
+     * encoding performs on the provided value before serialization. Options are:
+     *  * 'none' - no encoding. When used, the cookie value must be a string. This is the default value.
+     *  * 'base64' - string value is encoded using Base64.
+     *  * 'base64json' - object value is JSON-stringified then encoded using Base64.
+     *  * 'form' - object value is encoded using the x-www-form-urlencoded method.
+     *  * 'iron' - Encrypts and sign the value using iron.
+     */
+    encoding?: 'none' | 'base64' | 'base64json' | 'form' | 'iron';
+    /**
+     * an object used to calculate an HMAC for cookie integrity validation. This does not provide privacy, only a mean to verify that the cookie value was generated by the server. Redundant when 'iron' encoding is used. Options are:
+     *  * integrity - algorithm options. Defaults to require('iron').defaults.integrity.
+     *  * password - password used for HMAC key generation (must be at least 32 characters long).
+     */
+    sign?: {
+        integrity?: object;  // TODO make iron definitions and getting typing from iron. Needs review!
+        password: string;
+    };
+    /** password used for 'iron' encoding (must be at least 32 characters long). */
+    password?: string;
+    /** options for 'iron' encoding. Defaults to require('iron').defaults. */
+    iron?: object;  // TODO make iron definitions and getting typing from iron. Needs review!
```

Comment:
```
declare module 'iron' {
    export interface ISomething {
        saltBits: number;
        algorithm: string;
        iterations: number;
        minPasswordlength: number;
    }

    export interface ISealOptions {
        encryption: ISomething;
        integrity: ISomething;

        ttl: number;
        timestampSkewSec: number;
        localtimeOffsetMsec: number;
    }

    export const defaults: ISealOptions;

    export const algorithms: {
        'aes-128-ctr': {
            keyBits: number;
            ivBits: number;
        };
        'aes-256-cbc': {
            keyBits: number;
            ivBits: number;
        };
        'sha256': {
            keyBits: number;
        };
    };

    export const macFormatVersion: string;
    export const macPrefix: string;

    export interface IGenerateKeyOptions extends Pick<ISomething, 'algorithm' | 'iterations' | 'minPasswordlength'> {
        saltBits?: number;
        salt?: string;
        iv?: string;
    }

    export interface IKey {
        key: Buffer;
        salt: string;
        iv: string;
    }

    export function generateKey (password: string, options: IGenerateKeyOptions): Promise<IKey>;
    export function encrypt (password: string, options: IGenerateKeyOptions, data: string): Promise<{ data: Buffer, key: IKey }>;
    export function decrypt (password: string, options: IGenerateKeyOptions, data: string): Promise<Buffer>;

    export interface IHMacResult {
        digest: string;
        salt: string;
    }

    export function hmacWithPassword(password: string, options: IGenerateKeyOptions, data: string): Promise<IHMacResult>;

    export function seal (obj: object, password: string, options: ISealOptions): Promise<string>;
    export function unseal (data: string, password: string, options: ISealOptions): Promise<object>;
}
```

no warranties for correctness.

Reply:
Nice! Thanks @SimonSchick. Is it your code or did you get it from another place?

If it's new, and looks good comparing to [Iron js](https://github.com/hueniverse/iron/blob/master/lib/index.js) code, is not better to make a new PR creating the ```/types/iron``` ? Or should I put this code inside our hapi definitions?

Reply:
I just figured I would post it here, I didn't have time to PR these typings yet, feel free to PR them yourself.

Reply:
And yes, I wrote those typings.

Reply:
Commit 62ae76a. Thanks @SimonSchick